lnd_test.go

```diff
@@ -1029,6 +1029,7 @@ func testSingleHopInvoice(net *networkHarness, t *harnessTest) {
 		PaymentHash: invoiceResp.RHash,
 		Dest:        net.Bob.PubKey[:],
 		Amt:         paymentAmt,
+		MaxFee:      int64(lnwire.MaxPaymentMSat),
```

Comment:
For the sake of minimizing the diff: I'm not sure it's necessary to actually specify this value. If it is (due to the way the feature has been implemented), then it should be amended, as we shouldn't _force_ users to always specify a `MaxFee` parameter. Instead, it should strictly be optional.

Reply:
Originally this value was a pointer which could be left `nil` for no fee limit. After this conversation: https://github.com/lightningnetwork/lnd/pull/320#discussion_r139392829 , this change was made because it seems that proto3 provides no way to distinguish between unset singular fields and fields that were explicitly set to 0. This would be resolved by bringing back the `oneof` field.